lib/windows/library.py

```diff
@@ -1174,7 +1191,22 @@ def showPanelClicked(self):
                 self.processCommand(opener.handleOpen(subitems.ShowWindow, media_item=mli.dataSource, parent_list=self.showPanelControl))
             updateWatched = True
         elif self.section.TYPE == 'movie':
-            self.processCommand(opener.handleOpen(preplay.PrePlayWindow, video=mli.dataSource, parent_list=self.showPanelControl))
+            datasource = mli.dataSource
+            if datasource.isDirectory():
+                cls = self.section.__class__
+                section = cls(self.section.data, self.section.initpath, self.section.server, self.section.container)
+                sectionId = section.key
+                if not sectionId.isdigit():
+                    sectionId = section.getLibrarySectionId()
+
+                section.set('librarySectionID', sectionId)
+                section.key = datasource.key
+                section.title = datasource.title
+
+                self.processCommand(opener.handleOpen(LibraryWindow, windows=self._windows, default_window=self._next, section=section, filter_=self.filter, subDir=True))
+                self.librarySettings.setItemType(self.librarySettings.getItemType())
```

Comment:
I think this can be simplified as below - it seemed to work fine with the change:
```suggestion
            if datasource.isDirectory():
                cls = self.section.__class__
                section = cls(datasource.data, datasource.key, self.section.server, datasource.container)
                sectionId = section.key if section.key.isdigit() else section.getLibrarySectionId()

                section.set('librarySectionID', sectionId)

                self.processCommand(opener.handleOpen(LibraryWindow, windows=self._windows, default_window=self._next, section=section, subDir=True))
                self.librarySettings.setItemType(self.librarySettings.getItemType())
```

Reply:
In the above, the `section`'s `key` and `title` are no longer set.  It's been a long time since I've looked at this code but I think this had to do with entering a directory that was contained within a directory.

Reply:
I'm not sure what you mean. When you pass `datasource.data` to the `LibrarySection` constructor, the key and title are set to the key and title of the `datasource.data` (which is the raw data that initialized `datasource`). I tested it with a directory within a directory and it seemed to work fine (once the other folder reloading fix was added).